api/src/data.py

```diff
@@ -149,9 +201,70 @@ def cancel_job(job_id):
     return job.get_status()
 
 
+class FetchJobStatusResponseModel(BaseModel):
+    id: str
+    created_at: datetime
+    enqueued_at: Optional[datetime]
+    started_at: Optional[datetime]
+    status: RunStatusSchema
+    job_error: Optional[str]
+    result: Optional[dict]
+
+
+@router.get(
+    "/job/{uuid}/{job_string}",
+    response_model=FetchJobStatusResponseModel
+)
+def job_status(uuid: str, job_string: str):
+    """
+    If a job exists, returns the full job data.
+    """
+    job_id = f"{uuid}_{job_string}"
+
+    job = q.fetch_job(job_id)
+
+    # It is None, since no job exists
+    if not job:
+        return Response(
+            status_code=status.HTTP_404_NOT_FOUND,
+            content=(
+                f"No job data found for uuid = {uuid}/{job_string}.\n"
+                "This job may have existed, but cache expired."
+            ),
+        )
+
+    response = {
+        "id": job_id,
+        "created_at": job.created_at,
+        "enqueued_at": job.enqueued_at,
+        "started_at": job.started_at,
+        "status": job.get_status(),
+        "job_error": job.exc_info,
+        "result": job.result,
+    }
+    return response
+
+
+# TODO Use instead of Optional Dict|None (which doesn't generate swagger)
+class JobCreateOptions(BaseModel):
```

Comment:
Did not muster the courage to mess with the create `job` fn below, but this is a start- we can try to add it so that swagger displays better documentation/sample, and for developer happiness, of course.